src/main/java/dev/meyi/noscamspam/events/NSSChatHandler.java

```diff
@@ -29,14 +30,12 @@
       "^-----------------------------\\n(\\[.*] )?(.*) has invited you to join their party!\\nYou have 60 seconds to accept. Click here to join!\\n-----------------------------$");
   boolean firstJoin = true;
 
-  @SubscribeEvent
+  @SubscribeEvent(priority = EventPriority.HIGHEST)
   public void onPartyMessage(ClientChatReceivedEvent event) {
     if (!NoScamSpam.config.apiKey.equals("")) {
       Matcher m = party.matcher(event.message.getUnformattedText());
       if (m.find()) {
-        if (UserCheck.shouldBlockUser(m.group(2).toLowerCase())) {
-          event.setCanceled(true);
-        }
+        UserCheck.shouldBlockUser(m.group(2).toLowerCase(), event);
```

Comment:
If the `shouldBlockUser` method is changed to `void`, the name needs to be changed as well, as we aren't expecting the method to return anything.

Reply:
It can also be moved straight into the event like posted in your first comment